content/guides/box-mcp/self-hosted/index.md

```diff
@@ -0,0 +1,460 @@
+---
+rank: 1
+related_endpoints: []
+related_guides: []
+required_guides: []
+---
+
+# Self-hosted Box MCP Server
+
+The [Self-hosted Box MCP Server](https://github.com/box-community/mcp-server-box.git) is a Python project that integrates with the Box API to perform various operations such as file search, text extraction, AI-based querying, and data extraction. It leverages the Box Python Next Gen SDK library and provides a set of tools to interact with Box files and folders.
+
+## Installation
+
+### Prerequisites
+
+- Python 3.13 or higher
+- Box Platform app credentials (Client ID, Client Secret)
+
+Follow the steps from this section to set up the self-hosted Box MCP Server.
+
+1. Clone the repository:
+
+```sh
+git clone https://github.com/box-community/mcp-server-box.git
+cd mcp-server-box
+```
+
+2. Install `uv` if it's not installed on your machine:
+
+<Tabs>
+  <Tab title='MacOS/Linux'>
+    ```sh
+      curl -LsSf https://astral.sh/uv/install.sh | sh
+    ```
+  </Tab>
+  <Tab title='Windows'>
+    ```powershell
+      powershell -ExecutionPolicy ByPass -c "irm https://astral.sh/uv/install.ps1 | iex"
+    ```
+  </Tab>
+</Tabs>
+
+Restart your terminal afterwards to ensure that the `uv` command gets picked up.
+    
+3. Create and set up our project:
+
+<Tabs>
+  <Tab title='MacOS/Linux'>
+
+  ```sh
+    # Create virtual environment and activate it
+    uv venv
+    source .venv/bin/activate
+
+    # Lock the dependencies
+    uv lock
+  ```
+
+  </Tab>
+
+  <Tab title='Windows'>
+
+  ```sh
+    # Create virtual environment and activate it
+    uv venv
+    .venv\Scripts\activate
+
+    # Lock the dependencies
+    uv lock
+  ```
+
+  </Tab>
+</Tabs>
+
+4. Create a `.env` file in the root directory and fill your Box Platform app credentials:
+
+```.env
+BOX_CLIENT_ID=your_client_id
+BOX_CLIENT_SECRET=your_client_secret
+```
+
+You can also watch a video tutorial and see example usage of Box MCP tools.
+
+<iframe width="100%" height="500" src="https://www.youtube.com/embed/h109CMywlVQ?si=0EIRYyczpxfuidVU" title="Box self-hosted MCP Server" frameborder="0" allow="accelerometer; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
+
+## Running Box MCP Server locally
+
+To start the Box MCP Server, run the following command:
+
+```sh
+uv --directory /Users/USER_NAME/PATH_TO_PROJECT/mcp-server-box run src/mcp_server_box.py
+```
+
+Update the path so it reflects your local setup.
+
+### Use Cursor as the Box MCP client
+
+Prerequisites:
+
+- Download [Cursor desktop app](https://www.cursor.com/)
+
+Follow these instructions to start using Box MCP Sever with Cursor:
+
+1. Open the Cursor app.
+2. Click the cog icon located in the right-top corner to open settings.
```

Comment:
```suggestion
2. Click the cog icon to open settings.
```
A11y comment: Remove the direction indicators unless they are crucial and the user won't be able to understand the instruction without them.